includes/admin/class-wc-rest-payments-charges-controller.php

```diff
@@ -54,4 +63,64 @@ public function get_charge( $request ) {
 
 		return rest_ensure_response( $charge );
 	}
+
+	/**
+	 * Generates a charge-like object from an order.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 *
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function generate_charge_from_order( $request ) {
+		$order_id = $request['order_id'];
+		$order    = wc_get_order( $order_id );
+
+		if ( false === $order ) {
+			return new WP_Error( 'wcpay_missing_order', __( 'Order not found', 'woocommerce-payments' ), [ 'status' => 404 ] );
+		}
+
+		$amount          = (int) $order->get_total() * 100;
```

Comment:
👋 , sorry for jumping here as I am troubleshooting other issues @eduardoumpierre and @tpaksu 

This will result in the wrong `$amount` for decimal currencies such as `JPY`. 

May use `WC_Payments_Utils::interpret_stripe_amount` for this purpose. 

Note: Japanese merchants are supported by WooPayments at the moment https://woocommerce.com/payments/japan/

Reply:
This piece of code was merged on March, Japanese merchants were enabled on August, according to paJDYF-9bt-p2. The fix you mention here might have already been applied in that project, or in a PR after the expansion. This piece of code may be obsolete by now. Thanks for the heads up! If the problem still exists, can you create an issue for that?

Reply:
Well, @eduardoumpierre I've digged up the code a bit, and this is for the dummy charge we create for displaying the blocked order transactions. And @htdat is right, it'll show the ZDC amounts multiplied with 100. We may need to create an issue for that.

Reply:
The correct util would be `WC_Payments_Utils::prepare_amount` (`interpret_stripe_amount` is for the reversed way to translate value from Stripe API)

Created issue: https://github.com/Automattic/woocommerce-payments/issues/7583